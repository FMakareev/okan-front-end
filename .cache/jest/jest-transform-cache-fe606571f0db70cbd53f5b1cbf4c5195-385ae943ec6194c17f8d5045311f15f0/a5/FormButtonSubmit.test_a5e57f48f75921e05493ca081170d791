e08d7f52391df66575b8b8bf506e4eb4
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

var _StyledThemeProvider = require("../../styles/StyledThemeProvider");

var _FormButtonSubmit = require("./FormButtonSubmit");

// import 'jest-styled-components'; // этот импорт нужен чтобы в снэпшоты попали css стили которые генерируются при рендере компонента
test('FormButtonSubmit: Рендерится без ошибок', function () {
  var output = _reactTestRenderer["default"].create(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_FormButtonSubmit.FormButtonSubmit, {
    children: 'Button'
  }))).toJSON();

  expect(output).toMatchSnapshot();
});
test('FormButtonSubmit: Рендерится прелоадер если isLoading true', function () {
  var output = _reactTestRenderer["default"].create(_react["default"].createElement(_StyledThemeProvider.StyledThemeProvider, null, _react["default"].createElement(_FormButtonSubmit.FormButtonSubmit, {
    children: 'Button',
    isLoading: true
  })));

  expect(output).toMatchSnapshot();
}); // test('FormButtonSubmit: Рендерится с иконкой слева', () => {
//   const output = renderer.create(
//     <StyledThemeProvider>
//       <FormButtonSubmit children={'Button'} variant={'large'} size={'large'} />
//     </StyledThemeProvider>,
//   );
//   expect(output).toMatchSnapshot();
// });
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkZvcm1CdXR0b25TdWJtaXQudGVzdC5qcyJdLCJuYW1lcyI6WyJ0ZXN0Iiwib3V0cHV0IiwicmVuZGVyZXIiLCJjcmVhdGUiLCJ0b0pTT04iLCJleHBlY3QiLCJ0b01hdGNoU25hcHNob3QiXSwibWFwcGluZ3MiOiI7Ozs7QUFBQTs7QUFDQTs7QUFHQTs7QUFDQTs7QUFIQTtBQUtBQSxJQUFJLENBQUMseUNBQUQsRUFBNEMsWUFBTTtBQUNwRCxNQUFNQyxNQUFNLEdBQUdDLDhCQUNaQyxNQURZLENBRVgsZ0NBQUMsd0NBQUQsUUFDRSxnQ0FBQyxrQ0FBRDtBQUFrQixJQUFBLFFBQVEsRUFBRTtBQUE1QixJQURGLENBRlcsRUFNWkMsTUFOWSxFQUFmOztBQU9BQyxFQUFBQSxNQUFNLENBQUNKLE1BQUQsQ0FBTixDQUFlSyxlQUFmO0FBQ0QsQ0FURyxDQUFKO0FBV0FOLElBQUksQ0FBQyw0REFBRCxFQUErRCxZQUFNO0FBQ3ZFLE1BQU1DLE1BQU0sR0FBR0MsOEJBQVNDLE1BQVQsQ0FDYixnQ0FBQyx3Q0FBRCxRQUNFLGdDQUFDLGtDQUFEO0FBQWtCLElBQUEsUUFBUSxFQUFFLFFBQTVCO0FBQXNDLElBQUEsU0FBUztBQUEvQyxJQURGLENBRGEsQ0FBZjs7QUFLQUUsRUFBQUEsTUFBTSxDQUFDSixNQUFELENBQU4sQ0FBZUssZUFBZjtBQUNELENBUEcsQ0FBSixDLENBU0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XHJcbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcclxuLy8gaW1wb3J0ICdqZXN0LXN0eWxlZC1jb21wb25lbnRzJzsgLy8g0Y3RgtC+0YIg0LjQvNC/0L7RgNGCINC90YPQttC10L0g0YfRgtC+0LHRiyDQsiDRgdC90Y3Qv9GI0L7RgtGLINC/0L7Qv9Cw0LvQuCBjc3Mg0YHRgtC40LvQuCDQutC+0YLQvtGA0YvQtSDQs9C10L3QtdGA0LjRgNGD0Y7RgtGB0Y8g0L/RgNC4INGA0LXQvdC00LXRgNC1INC60L7QvNC/0L7QvdC10L3RgtCwXHJcblxyXG5pbXBvcnQgeyBTdHlsZWRUaGVtZVByb3ZpZGVyIH0gZnJvbSAnLi4vLi4vc3R5bGVzL1N0eWxlZFRoZW1lUHJvdmlkZXInO1xyXG5pbXBvcnQgeyBGb3JtQnV0dG9uU3VibWl0IH0gZnJvbSAnLi9Gb3JtQnV0dG9uU3VibWl0JztcclxuXHJcbnRlc3QoJ0Zvcm1CdXR0b25TdWJtaXQ6INCg0LXQvdC00LXRgNC40YLRgdGPINCx0LXQtyDQvtGI0LjQsdC+0LonLCAoKSA9PiB7XHJcbiAgY29uc3Qgb3V0cHV0ID0gcmVuZGVyZXJcclxuICAgIC5jcmVhdGUoXHJcbiAgICAgIDxTdHlsZWRUaGVtZVByb3ZpZGVyPlxyXG4gICAgICAgIDxGb3JtQnV0dG9uU3VibWl0IGNoaWxkcmVuPXsnQnV0dG9uJ30gLz5cclxuICAgICAgPC9TdHlsZWRUaGVtZVByb3ZpZGVyPixcclxuICAgIClcclxuICAgIC50b0pTT04oKTtcclxuICBleHBlY3Qob3V0cHV0KS50b01hdGNoU25hcHNob3QoKTtcclxufSk7XHJcblxyXG50ZXN0KCdGb3JtQnV0dG9uU3VibWl0OiDQoNC10L3QtNC10YDQuNGC0YHRjyDQv9GA0LXQu9C+0LDQtNC10YAg0LXRgdC70LggaXNMb2FkaW5nIHRydWUnLCAoKSA9PiB7XHJcbiAgY29uc3Qgb3V0cHV0ID0gcmVuZGVyZXIuY3JlYXRlKFxyXG4gICAgPFN0eWxlZFRoZW1lUHJvdmlkZXI+XHJcbiAgICAgIDxGb3JtQnV0dG9uU3VibWl0IGNoaWxkcmVuPXsnQnV0dG9uJ30gaXNMb2FkaW5nIC8+XHJcbiAgICA8L1N0eWxlZFRoZW1lUHJvdmlkZXI+LFxyXG4gICk7XHJcbiAgZXhwZWN0KG91dHB1dCkudG9NYXRjaFNuYXBzaG90KCk7XHJcbn0pO1xyXG5cclxuLy8gdGVzdCgnRm9ybUJ1dHRvblN1Ym1pdDog0KDQtdC90LTQtdGA0LjRgtGB0Y8g0YEg0LjQutC+0L3QutC+0Lkg0YHQu9C10LLQsCcsICgpID0+IHtcclxuLy8gICBjb25zdCBvdXRwdXQgPSByZW5kZXJlci5jcmVhdGUoXHJcbi8vICAgICA8U3R5bGVkVGhlbWVQcm92aWRlcj5cclxuLy8gICAgICAgPEZvcm1CdXR0b25TdWJtaXQgY2hpbGRyZW49eydCdXR0b24nfSB2YXJpYW50PXsnbGFyZ2UnfSBzaXplPXsnbGFyZ2UnfSAvPlxyXG4vLyAgICAgPC9TdHlsZWRUaGVtZVByb3ZpZGVyPixcclxuLy8gICApO1xyXG4vLyAgIGV4cGVjdChvdXRwdXQpLnRvTWF0Y2hTbmFwc2hvdCgpO1xyXG4vLyB9KTtcclxuIl19