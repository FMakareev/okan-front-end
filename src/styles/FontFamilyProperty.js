/**
 * @params {string} fontFamily - string name of the font name property in the theme
 * @params {object} theme
 * */
export const FontFamilyProperty = ({ fontFamily, theme }) => {
  try {
    if (fontFamily && theme.fontFamily) {
      return `font-family: '${theme.fontFamily[fontFamily]}';`;
    }
  } catch (e) {
    console.error(e);
  }
  return undefined;
};
export default FontFamilyProperty;
