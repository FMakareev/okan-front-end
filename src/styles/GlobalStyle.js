import { createGlobalStyle } from 'styled-components'


export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
  }
`;
