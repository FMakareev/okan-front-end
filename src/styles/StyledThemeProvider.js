import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';

/** Variants */
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { ButtonComment } from './variants/buttonComment';
import { TextVariant } from './variants/textVariant';
import { InputSize } from '@lib/styles/variants/InputSize';
import { InputVariant } from '@lib/styles/variants/InputVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#E5E5E5',
  color2: '#DEDEDE',
  color3: '#4F4F4F',
  color4: '#848484',
  color5: '#007FAF',
  color6: '#006699',
  color7: '#00649C',
  color8: '#007CC4',
  color9: '#F97E63',
  color10: '#AEAEAE',
  color11: '#333333',
  color12: '#DF4624',
  color13: '#f2f2f2',
  color14: '#fff3c1',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['1024px', '1200px', '1440px'];

const boxShadow = [
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 4px 8px 0px rgba(138,138,138,0.5)',
  '2px 2px 4px 0px rgba(0,127,175,1)',
];

const textAlign = ['left', 'center', 'right'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        buttonComment: ButtonComment,
        text: TextVariant,
        inputSize: InputSize,
        inputVariant: InputVariant,
      },
      fontFamily: {
        primary300: 'Museo Sans 300',
        primary500: 'Museo Sans 500',
        primary700: 'Museo Sans 700',
        secondary: 'Circe Regular',
        secondaryBold: 'Circe Bold',
      },
      textAlign: textAlign,
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
