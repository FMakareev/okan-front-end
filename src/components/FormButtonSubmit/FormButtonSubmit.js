import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { color } from 'styled-system';

/** View */
import Flex from '../Flex/Flex';
import ButtonWithImage from '../ButtonWithImage/ButtonWithImage';
import SmallPreloader from '../SmallPreloader/SmallPreloader';

/** Image */
import { SvgPlay } from '../Icons/SvgPlay';

/** style property */
import { LineHeightProperty } from '../../styles/styleProperty/LineHeightProperty';
import { FontFamilyProperty } from '../../styles/styleProperty/FontFamilyProperty';
import { FontSizeProperty } from '../../styles/styleProperty/FontSizeProperty';

const Error = styled.span`
  ${props => FontFamilyProperty({ ...props, fontFamily: 'secondary' })};
  ${props => FontSizeProperty({ ...props, fontSize: '16px' })};
  ${props => LineHeightProperty({ ...props, lineHeight: '18px' })};
  ${props => color({ ...props, color: 'color12' })};
`;

export const FormButtonSubmit = ({ disabled, children, ml, error, isLoading }) => {
  const buttonIconRender = isLoading ? <SmallPreloader /> : SvgPlay();

  return (
    <Flex flexDirection={'column'} justifyContent={'center'}>
      {error && <Error>{error}</Error>}
      <ButtonWithImage
        type="submit"
        disabled={disabled}
        variant={'large'}
        size={'large'}
        rightIcon={buttonIconRender}
        ml={ml}>
        {children}
      </ButtonWithImage>
    </Flex>
  );
};

FormButtonSubmit.propTypes = {
  /** disabled input */
  disabled: PropTypes.bool,
  /** name button */
  children: PropTypes.string,
  /** css property: margin-left */
  ml: PropTypes.number,
  /** error */
  error: PropTypes.string,
};

export default FormButtonSubmit;

// &:disabled {
//   cursor: default;
//   border-color: ${props => CreateColor(color(props).borderColor)
//       .darken(0.7)
//       .hex()};
// }
