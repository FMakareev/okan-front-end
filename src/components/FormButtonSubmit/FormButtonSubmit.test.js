import React from 'react';
import renderer from 'react-test-renderer';
// import 'jest-styled-components'; // этот импорт нужен чтобы в снэпшоты попали css стили которые генерируются при рендере компонента

import { StyledThemeProvider } from '../../styles/StyledThemeProvider';
import { FormButtonSubmit } from './FormButtonSubmit';

test('FormButtonSubmit: Рендерится без ошибок', () => {
  const output = renderer
    .create(
      <StyledThemeProvider>
        <FormButtonSubmit children={'Button'} />
      </StyledThemeProvider>,
    )
    .toJSON();
  expect(output).toMatchSnapshot();
});

test('FormButtonSubmit: Рендерится прелоадер если isLoading true', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <FormButtonSubmit children={'Button'} isLoading />
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});

// test('FormButtonSubmit: Рендерится с иконкой слева', () => {
//   const output = renderer.create(
//     <StyledThemeProvider>
//       <FormButtonSubmit children={'Button'} variant={'large'} size={'large'} />
//     </StyledThemeProvider>,
//   );
//   expect(output).toMatchSnapshot();
// });
