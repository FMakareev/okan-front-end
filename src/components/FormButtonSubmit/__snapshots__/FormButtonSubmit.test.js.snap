// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormButtonSubmit: Рендерится без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 wterhv-0 fCdjsl"
  >
    <button
      className="sc-8daz6k-1 IebJL mxzgdr-0 gVQTvs"
      size="large"
      type="submit"
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 eZMgmL"
        width="100%"
      >
        <div
          className="sc-1jxwge6-0 wterhv-0 kyYPav"
        >
          Button
        </div>
        <div
          className="sc-1jxwge6-0 wterhv-0 fuRcbY"
        >
          <svg
            fill="inherit"
            height="1.250em"
            viewBox="0 0 16 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M15.1351 8.58297L2.56862 0.197654C2.19712 -0.0365882 1.73582 0.00243778 1.38742 0.00243778C-0.00618175 0.00243778 4.78129e-07 1.14194 4.78129e-07 1.43063V18.5693C4.78129e-07 18.8134 -0.00610045 19.9976 1.38742 19.9976C1.73582 19.9976 2.1972 20.0364 2.56862 19.8023L15.135 11.417C16.1665 10.767 15.9882 9.99997 15.9882 9.99997C15.9882 9.99997 16.1665 9.23289 15.1351 8.58297Z"
              fill="inherit"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`FormButtonSubmit: Рендерится прелоадер если isLoading true 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 wterhv-0 fCdjsl"
  >
    <button
      className="sc-8daz6k-1 IebJL mxzgdr-0 gVQTvs"
      size="large"
      type="submit"
    >
      <div
        className="sc-1jxwge6-0 wterhv-0 eZMgmL"
        width="100%"
      >
        <div
          className="sc-1jxwge6-0 wterhv-0 kyYPav"
        >
          Button
          <div
            className="sc-1jxwge6-0 sc-181m7t4-0 gCxfwX"
          >
            <div
              className="sc-181m7t4-1 jGYmHw"
            />
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 wterhv-0 fuRcbY"
        >
          <svg
            fill="inherit"
            height="1.250em"
            viewBox="0 0 16 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M15.1351 8.58297L2.56862 0.197654C2.19712 -0.0365882 1.73582 0.00243778 1.38742 0.00243778C-0.00618175 0.00243778 4.78129e-07 1.14194 4.78129e-07 1.43063V18.5693C4.78129e-07 18.8134 -0.00610045 19.9976 1.38742 19.9976C1.73582 19.9976 2.1972 20.0364 2.56862 19.8023L15.135 11.417C16.1665 10.767 15.9882 9.99997 15.9882 9.99997C15.9882 9.99997 16.1665 9.23289 15.1351 8.58297Z"
              fill="inherit"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;
